src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -296,12 +300,49 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
                 }
                 services.TryAddSingleton<FunctionsScaleManager>();
+
+                services.AddHostOverrides();
             });
 
             RegisterFileProvisioningService(builder);
             return builder;
         }
 
+        public static void AddHostOverrides(this IServiceCollection services)
+        {
+            // Custom implementations that the Host overrides
+            services.AddSingleton<ScheduleMonitor, AzureStorageScheduleMonitor>();
+            services.AddSingleton<IDistributedLockManager>(provider => AddLockManager(provider));
+        }
+
+        public static IDistributedLockManager AddLockManager(IServiceProvider provider)
+        {
+            try
+            {
+                var azureStorageProvider = provider.GetRequiredService<IAzureStorageProvider>();
+                var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                var container = azureStorageProvider.GetBlobContainerClient();
+
+                if (container != null)
+                {
+                    return new BlobLeaseDistributedLockManager(loggerFactory, azureStorageProvider);
+                }
+            }
+            catch (Exception ex)
+            {
+                var str = ex.ToString();
+            }
+
+            return new InMemoryDistributedLockManager();
```

Comment:
Is this fallback to the inmemory manager just for testing? Is it possible we'll hit this codepath in prod, and the fact that its silently falling back will be confusing? Should we emit a log entry about the fallback?

Reply:
This is intended behavior for cases where Blob storage isn't available (i.e. a placeholder that is warmed up will use this InMemoryDistributedLockManager. Don't think we should log here because it is intended and may just cause more noise.